vnext/Microsoft.ReactNative.Managed.CodeGen.UnitTests/Microsoft.ReactNative.Managed.CodeGen.UnitTests.csproj

```diff
@@ -0,0 +1,79 @@
+
+      This seems to make nuget and msbuild agree with one another on the right version..
+    -->
+    <ImplicitPackageReferenceVersion Remove="Microsoft.NETCore.App" TargetFrameworkVersion="2.2" DefaultVersion="2.2.0" LatestVersion="2.2.8"/>
+    <ImplicitPackageReferenceVersion Include="Microsoft.NETCore.App" TargetFrameworkVersion="2.2" DefaultVersion="2.2.8" LatestVersion="2.2.8"/>
+  </ItemGroup>
+  
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
+    <PlatformTarget>x64</PlatformTarget>
+    <RuntimeIdentifiers>win-x64</RuntimeIdentifiers>
+    <IsPackable>false</IsPackable>
+    <Platforms>x64;x86;ARM;ARM64</Platforms>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <AvailableItemName Include="Lkg" />
+    <Lkg Include="CodeGen\Lkg\*.lkg" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.6.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.6.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
+    <PackageReference Include="MSTest.TestAdapter" Version="2.1.0" />
+    <PackageReference Include="MSTest.TestFramework" Version="2.1.0" />
+    <PackageReference Include="coverlet.collector" Version="1.2.0" />
+  </ItemGroup>
+
+   <!-- Ensure the win10 sdk binaries loaded by the unittest match our apps sdk versions -->
+  <PropertyGroup>
+    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersionEncoded>$(WindowsTargetPlatformVersion.Replace('.', '_'))</WindowsTargetPlatformVersionEncoded>
+    <DefineConstants>$(p:DefineConstants);win10SdkVersion$(WindowsTargetPlatformVersionEncoded)</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Folder Include="CodeGen\Lkg\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.ReactNative.Managed.CodeGen\Microsoft.ReactNative.Managed.CodeGen.csproj" />
+  </ItemGroup>
+
+  <Target Name="CopyLkgFiles" AfterTargets="Build">
+    <Copy SourceFiles="@(Lkg)" DestinationFiles="@(Lkg->'$(OutDir)Lkg\%(Filename)%(Extension)')" />
+
+    <ItemGroup>
+      <FileWrites Include="@(Lkg->'$(OutDir)Lkg\%(Filename)%(Extension)')" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="EnsureRuntimeDependenciesAreBuilt" AfterTargets="Build">
+    <ItemGroup>
+      <_ReactNativeManagedCodeGenUnitTestRuntimeDependencies Include="..\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj" />
+      <_ReactNativeManagedCodeGenUnitTestRuntimeDependencies Include="..\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed.csproj" />
+    </ItemGroup>
+    <MSBuild
```

Comment:
beware of this - you could end up with more than one MSBuild process trying to build the same project simultaneously and have spurious/intermittent build breaks due to locked files

Reply:
I spent a lot of time in the msbuild structured log viewer to make sure that these msbuild calls match the global properties of a project reference so that msbuild-target-caching should kick in.

Reply:
@dannyvv I was mostly referring to multiproc builds and having a solution that has a ProjectReference = Microsoft.ReactNative.vcxproj, as well as having this MSBuild target.
In that case, there won't be any coordination between the main msbuild process and the MSBuild target's and you can end up with one stomping over the other